classes/class-wc-connect-taxjar-integration.php

```diff
@@ -268,10 +331,21 @@ public function _log( $message ) {
 	public function _error( $message ) {
 		$formatted_message = is_scalar( $message ) ? $message : json_encode( $message );
 
-		//ignore error messages caused by customer input
+		// display these error messages directly to the user
+		// this error will be displayed on both cart and checkout pages
+		// and will block any further action
 		$state_zip_mismatch = false !== strpos( $formatted_message, 'to_zip' ) && false !== strpos( $formatted_message, 'is not used within to_state' );
 		$invalid_postcode = false !== strpos( $formatted_message, 'isn\'t a valid postal code for' );
-		if ( ! is_admin() && ( $state_zip_mismatch || $invalid_postcode ) ) {
+
+		// We should only block checkout if the user can edit the address on cart
+		$user_can_edit_address_on_cart = 'yes' === get_option( 'woocommerce_enable_shipping_calc' );
+		$should_not_add_wc_error_notice = ! $user_can_edit_address_on_cart && is_checkout();
```

Comment:
`$should_not_add_wc_error_notice` is confusing. Please change it to `$should_add_wc_error_notice` so the logic is easier to follow.